src/Serilog/Context/LogContext.cs

```diff
@@ -132,33 +147,63 @@ static ImmutableStack<ILogEventEnricher> GetOrCreateEnricherStack()
             return enrichers;
         }
 
+#if ASYNCLOCAL
+       static ImmutableStack<ILogEventEnricher> Enrichers
+        {
+            get
+            {
+                return data.Value;
+            }
+            set
+            {
+                data.Value = GetContext(value);
+            }
+        }
+#else
         static ImmutableStack<ILogEventEnricher> Enrichers
         {
             get
             {
-
                 var data = CallContext.LogicalGetData(DataSlotName);
 
                 ImmutableStack<ILogEventEnricher> context;
+#if REMOTING
                 if (PermitCrossAppDomainCalls)
```

Comment:
Moved `PermitCrossAppDomainCalls` property to inside the REMOTING block at the bottom of the file.